### PR TITLE
Add example of `SMESolver.run_from_experiement` in the guide.

### PR DESCRIPTION
**Pull request by Ericgig**
**Description**
Add a simple example for `run_from_experiement` in the guide.
Also restore the link to the stochastic tutorial notebooks.

### CODE REVIEW

File: doc/guide/dynamics/dynamics-stochastic.rst [hunk 1]
```diff
@@ -130,16 +130,21 @@ Example
...
-where :math:`x` is the operator passed using ``m_ops``. The results are available
-in ``result.measurements``.
+where :math:`x` is the operator build from the ``sc_ops`` as
+
+.. math::
+
+    x_n = S_n + S_n^\dagger
+
+
+The results are available in ``result.measurements``.
```
Comment (pmenczel) [inline doc/guide/dynamics/dynamics-stochastic.rst:147]:
```suggestion
The results are available in ``result.measurement``.
```

File: doc/guide/dynamics/dynamics-stochastic.rst [hunk 2]
```diff
@@ -178,21 +183,52 @@ in ``result.measurements``.
...
+    solver = SMESolver(
+        H, sc_ops=[np.sqrt(KAPPA) * a],
+        options={"dt": 0.00125, "store_measurement": True,}
+    )
+
+    # Run the evolution, noise
+    recreated_solution = solver.run_from_experiment(
+        rho_0, tlist, stoc_solution.measurements[0],
+        e_ops=[H],
+        # The third parameter is a measurement, not Gaussian noise
```
Comment (pmenczel) [inline doc/guide/dynamics/dynamics-stochastic.rst:208]:
```suggestion
        # The third parameter is the measurement, not the Wiener increment
```
I find more easily understandable

File: doc/guide/dynamics/dynamics-stochastic.rst [hunk 3]
```diff
@@ -178,21 +183,52 @@ in ``result.measurements``.
...
+        measurement=True,
+    )
+
+This will recompute the states, expectation values and wiener increments for that trajectory.
+
+.. note::
+
+  The measurements in the result is computed from the state at the end of the time step.
+  However, when using ``run_from_experiment`` with measurement input, the state at the start of the time step is used.
```
Comment (pmenczel) [inline doc/guide/dynamics/dynamics-stochastic.rst:217]:
```suggestion
  The measurement in the result is by default computed from the state at the end of the time step.
  However, when using ``run_from_experiment`` with measurement input, the state at the start of the time step is used.
  To obtain the measurement at the start of the time step in the output of ``smesolve``, one may use the option ``{'store_measurement': 'start'}``.
```

File: doc/guide/dynamics/dynamics-stochastic.rst [hunk 4]
```diff
@@ -178,21 +183,52 @@ in ``result.measurements``.
...
+
+
+For other examples on :func:`qutip.solver.stochastic.smesolve`, see the
+notebooks available at `QuTiP Tutorials page <https://qutip.org/tutorials.html>`_:
+
+`heterodyne detection <https://nbviewer.org/urls/qutip.org/qutip-tutorials/tutorials-v5/time-evolution/015_smesolve-heterodyne.ipynb>`_,
+`inefficient detection <https://nbviewer.org/urls/qutip.org/qutip-tutorials/tutorials-v5/time-evolution/016_smesolve-inefficient-detection.ipynb>`_, and
+`feedback control <https://github.com/jrjohansson/reproduced-papers/blob/master/Reproduce-SIAM-JCO-46-445-2007-Mirrahimi.ipynb>`_.
```
Comment (pmenczel) [inline doc/guide/dynamics/dynamics-stochastic.rst:225]:
```suggestion
notebooks available on the `QuTiP Tutorials page <https://qutip.org/tutorials.html>`_:

* `Heterodyne detection <https://nbviewer.org/urls/qutip.org/qutip-tutorials/tutorials-v5/time-evolution/015_smesolve-heterodyne.ipynb>`_
* `Inefficient detection <https://nbviewer.org/urls/qutip.org/qutip-tutorials/tutorials-v5/time-evolution/016_smesolve-inefficient-detection.ipynb>`_
* `Feedback control <https://github.com/jrjohansson/reproduced-papers/blob/master/Reproduce-SIAM-JCO-46-445-2007-Mirrahimi.ipynb>`_
```

Fixed the formatting a bit. Why are we linking to a notebook in an external repo?

Comment (Ericgig) [reply doc/guide/dynamics/dynamics-stochastic.rst:225]: We had it in our tutorial even if it was not in our repo... It's no longer there so I will remove it.

**Review by pmenczel — APPROVED** [review]

**Pull request merged**